openquake/hazardlib/gsim/kotha_2019.py

```diff
@@ -350,10 +426,109 @@ def get_stddevs(self, C, stddev_shape, stddev_types, sites):
+    5.000   2.009611059381530   3.508361778014590   0.395686780409462    0.861356359764995   -1.208963806141110   0.228641145313669   -0.140697393026636   0.136942909437525   0.559951442166458   0.448040327766279   0.209658464567696   0.377238739334535    2.50349509    -0.39960740    0.34047678    3.05922169   -0.47921741    0.45274837
+    6.000   1.688862627964000   3.516737645416790   0.420264280758498    1.309501218469920   -1.214089541959290   0.237065702688919   -0.133212839440133   0.141314902795929   0.536431141593929   0.478894088472203   0.212035440719214   0.384111550789384    2.41685314    -0.38612724    0.33122324    2.89854557   -0.45396539    0.44059217
+    7.000   1.341373165722010   3.465765382316050   0.416036203766797    1.376730174631370   -1.277754681712860   0.267078137770598   -0.088623284188977   0.145375015287528   0.522666557851543   0.427622455537343   0.214045145419707   0.384907261550421    2.39868751    -0.38342544    0.32324116    2.75868843   -0.43197944    0.42858057
+    8.000   1.063213165919580   3.450720620500530   0.427534350267617    1.454576290963340   -1.323042041332830   0.291715072522946   -0.064115203294262   0.151582553207261   0.507211779986322   0.422147964378804   0.234785059413908   0.386735725945679    2.39997563    -0.38372495    0.32003214    2.70660001   -0.42378341    0.42381284
+    """)
+
+
+class KothaEtAl2019SERASlopeGeology(KothaEtAl2019SERA):
+    """
+    Adaptation of the Kotha et al. (2019) GMPE for use with slope and geology
+    in place of inferred/measured Vs30.
+    """
+    experimental = True
+
+    #: Required site parameter is not set
+    REQUIRES_SITES_PARAMETERS = set(("slope", "geology"))
+
+    #: Geological Units
+    GEOLOGICAL_UNITS = [b"CENOZOIC", b"HOLOCENE", b"MESOZOIC",
+                        b"PALEOZOIC", b"PLEISTOCENE", b"PRECAMBRIAN"]
+
+    def get_site_amplification(self, C, sites, imt):
+        """
+        Returns the site amplification term depending on whether the Vs30
+        is observed of inferred
+        """
+        C_AMP_FIXED = self.COEFFS_FIXED[imt]
+        C_AMP_RAND_INT = self.COEFFS_RANDOM_INT[imt]
+        C_AMP_RAND_GRAD = self.COEFFS_RANDOM_GRAD[imt]
+        ampl = np.zeros(sites.slope.shape)
+        geol_units = np.unique(sites.geology)
+        t_slope = np.copy(sites.slope)
+        t_slope[t_slope > 0.1] = 0.1
+        for geol_unit in geol_units:
+            idx = sites.geology == geol_unit
+            if geol_unit in self.GEOLOGICAL_UNITS:
+                # Supported geological unit - use the random effects model
+                v1 = C_AMP_FIXED["V1"] + C_AMP_RAND_INT[geol_unit.decode()]
+                v2 = C_AMP_FIXED["V2"] + C_AMP_RAND_GRAD[geol_unit.decode()]
+            else:
```

Comment:
Are you sure you want this? If the geology of a site is misspelt you will apply this without knowing it. Did you consider using an `unknown` geological category?

Reply:
It's difficult to decide where to put the check. I think it is more likely that the user might map an existing geological database to the input file, which could contain many units besides the ones listed here and not explicitly be classed as "unknown" per se, and in these cases the same fixed effects should be used as is the true "unknown" case. Rather than require the users to create such a mapping in which everything but the supported units is classed as "unknown" I would rather let it run on the broadest set geological units and leave it to the users to ensure that their data are correct.